Tests/NoFailedDependenciesConditionTests.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  ProcedureKit
+//
+//  Copyright © 2016 ProcedureKit. All rights reserved.
+//
+
+import XCTest
+import TestingProcedureKit
+@testable import ProcedureKit
+
+class NoFailedDependenciesConditionTests: ProcedureKitTestCase {
+
+    func test__procedure_with_no_dependencies_succeeds() {
+        procedure.attach(condition: NoFailedDependenciesCondition())
+        wait(for: procedure)
+        XCTAssertProcedureFinishedWithoutErrors()
+    }
+
+    func test__procedure_with_successful_dependency_succeeds() {
+        let dependency = TestProcedure()
+        procedure.add(dependency: dependency)
+        procedure.attach(condition: NoFailedDependenciesCondition())
+        wait(for: procedure, dependency)
+        XCTAssertProcedureFinishedWithoutErrors()
+    }
+
+    func test__prcoedure_with_cancelled_dependency_fails() {
+        let dependency = createCancellingProcedure()
+        procedure.add(dependency: dependency)
+        procedure.attach(condition: NoFailedDependenciesCondition())
+        wait(for: procedure, dependency)
+        XCTAssertProcedureCancelledWithErrors(count: 1)
```

Comment:
@swiftlyfalling - this test will fail, because the expectation will not get fulfilled, because the `DidFinishObserver` is not called.